src/co/elastic/NotificationManager.groovy

```diff
@@ -52,76 +52,81 @@ def analyzeFlakey(Map args = [:]) {
     def querySize = args.get('querySize', 500)
     def queryTimeout = args.get('queryTimeout', '20s')
     def disableGHComment = args.get('disableGHComment', false)
+
     def labels = 'flaky-test,ci-reported'
+    def boURL = getBlueoceanDisplayURL()
+    def flakyTestsWithIssues = [:]
+    def genuineTestFailures = []
 
     if (!flakyReportIdx?.trim()) {
       error 'analyzeFlakey: did not receive flakyReportIdx data'
     }
 
-    // Query only the test_name field since it's the only used and don't want to overkill the
-    // jenkins instance when using the toJSON step since it reads in memory the json response.
-    // for 500 entries it's about 2500 lines versus 8000 lines if no filter_path
-    def query = "/${flakyReportIdx}/_search?size=${querySize}&filter_path=hits.hits._source.test_name,hits.hits._index"
-    def flakeyTestsRaw = sendDataToElasticsearch(es: es,
-                                                 secret: secret,
-                                                 data: queryFilter(queryTimeout, flakyThreshold),
-                                                 restCall: query)
-    def flakeyTestsParsed = toJSON(flakeyTestsRaw)
+    // Only if there are test failures to analyse
+    if(testsErrors.size() > 0) {
```

Comment:
New conditional to skip if not test errors, this is the new change in addition to the above declaration of variables for the notifications